src/devices/plic.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+enum PLIC_REG {
+    INTR_PENDING = 0x400,
+    INTR_ENABLE = 0x800,
+    INTR_PRIORITY = 0x80000,
+    INTR_CLAIM_OR_COMPLETE = 0x80001,
+};
```

Comment:
May I know why we are using the plic address >>2 here for register indexing? Using the offset directly seems to be easier to reference to [plic spec](https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc).

Reply:
> May I know why we are using the plic address >>2 here for register indexing?

The memory address is byte-addressable originally, but when accessing the the MMIO memory region, `uint32_t` type is used, so `address >> 2` to convert to word-addressable. You can see the similar implementation for MMU in 'system.c'.

Reply:
Still confusing. What's the risk for a change like this?

```diff
--- a/src/devices/plic.h
+++ b/src/devices/plic.h
@@ -8,10 +8,10 @@
 #include <stdint.h>

 enum PLIC_REG {
-    INTR_PENDING = 0x400,
-    INTR_ENABLE = 0x800,
-    INTR_PRIORITY = 0x80000,
-    INTR_CLAIM_OR_COMPLETE = 0x80001,
+    INTR_PENDING = 0x1000,
+    INTR_ENABLE = 0x2000,
+    INTR_PRIORITY = 0x200000,
+    INTR_CLAIM_OR_COMPLETE = 0x200004,
 };

--- a/src/system.c
+++ b/src/system.c
@@ -37,11 +37,11 @@ enum SUPPORTED_MMIO {
     switch(io){                                                                     \
         case MMIO_PLIC:                                                             \
             IIF(rw)( /* read */                                                     \
-                mmio_read_val = plic_read(PRIV(rv)->plic, (addr & 0x3FFFFFF) >> 2); \
+                mmio_read_val = plic_read(PRIV(rv)->plic, (addr & 0x3FFFFFF));      \
                 plic_update_interrupts(PRIV(rv)->plic);                             \
                 return mmio_read_val;                                               \
                 ,    /* write */                                                    \
-                plic_write(PRIV(rv)->plic, (addr & 0x3FFFFFF) >> 2, val);           \
+                plic_write(PRIV(rv)->plic, (addr & 0x3FFFFFF), val);                \
                 plic_update_interrupts(PRIV(rv)->plic);                             \
                 return;                                                             \
             )                                                                       \
```

Reply:
Got it. The changes seem applicable and I'll adopt the changes.